components/raftstore-v2/src/operation/command/write/mod.rs

```diff
@@ -83,10 +90,67 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
     }
 
     pub fn propose_pending_command<T>(&mut self, ctx: &mut StoreContext<EK, ER, T>) {
-        if let Some(encoder) = self.raw_write_encoder_mut().take() {
+        if let Some(encoder) = self.simple_write_encoder_mut().take() {
             let (data, chs) = encoder.encode();
             let res = self.propose(ctx, data);
             self.post_propose_write(ctx, res, chs);
         }
     }
 }
+
+impl<EK: KvEngine, R> Apply<EK, R> {
+    #[inline]
+    pub fn apply_put(&mut self, cf: &str, key: &[u8], value: &[u8]) -> Result<()> {
+        util::check_key_in_region(key, self.region_state().get_region())?;
+        let res = if cf.is_empty() || cf == CF_DEFAULT {
+            // TODO: use write_vector
+            self.write_batch_or_default().put(key, value)
+        } else {
+            self.write_batch_or_default().put_cf(cf, key, value)
+        };
+        res.unwrap_or_else(|e| {
+            panic!(
+                "{:?} failed to write ({}, {}) {}: {:?}",
+                self.logger.list(),
+                log_wrappers::Value::key(key),
+                log_wrappers::Value::value(value),
+                cf,
+                e
+            );
+        });
+        Ok(())
+    }
+
+    #[inline]
+    pub fn apply_delete(&mut self, cf: &str, key: &[u8]) -> Result<()> {
+        util::check_key_in_region(key, self.region_state().get_region())?;
+        let res = if cf.is_empty() || cf == CF_DEFAULT {
+            // TODO: use write_vector
+            self.write_batch_or_default().delete(key)
+        } else {
+            self.write_batch_or_default().delete_cf(cf, key)
```

Comment:
Could `cf == CF_DEFAULT` use this path too?

Reply:
Yes, but rocksdb has optimization on default cf, which can make the write batch smaller.